File: android_kotlin_app/app/src/main/java/com/mindsilence/game/feature/game/DailyStats.kt
package com.mindsilence.game.feature.game // One calendar day's persisted training stats.

import androidx.compose.runtime.Immutable // List items are stable snapshots.
import java.time.LocalDate // Local calendar day, not a timestamp.

/** One day's recorded attempts: count, summed silence time, and best level reached. */
@Immutable // Highscore rows do not mutate in place.
data class DailyStats( // JSON fields: date, attempts, totalSeconds, bestLevel.
    val date: LocalDate, // Day key (device local).
    val attempts: Int, // Thoughts recorded that day.
    val totalSeconds: Int, // Sum of attempt lengths.
    val bestLevel: Int, // Max levelReached that day.
) // End DailyStats.

File: Android_maui_app/Domain/Models/DailyStats.cs
namespace MindSilence.Domain.Models;

public sealed record DailyStats(DateOnly Date, int Attempts, int TotalSeconds, int BestLevel)
{
	public static DailyStats ForFirstAttempt(DateOnly date, int levelReached, int totalSeconds) =>
		new(date, Attempts: 1, totalSeconds, levelReached);

	public DailyStats AddAttempt(int levelReached, int totalSeconds) => this with
	{
		Attempts = Attempts + 1,
		TotalSeconds = TotalSeconds + totalSeconds,
		BestLevel = Math.Max(BestLevel, levelReached),
	};
}
